packages/worker/src/worker.ts

```diff
@@ -733,21 +860,37 @@ export class Worker {
     });
   }
 
-  /**
-   * Start polling on tasks, completes after graceful shutdown due to receiving a shutdown signal
-   * or call to {@link shutdown}.
-   */
-  async run(): Promise<void> {
-    if (this.state !== 'INITIALIZED') {
-      throw new Error('Poller was aleady started');
-    }
-    this.state = 'RUNNING';
+  protected activity$(): Observable<void> {
+    return merge(this.activityPoll$(), this.activityFeedbackSubject.pipe(this.takeUntilState('DRAINING'))).pipe(
+      this.activityOperator(),
+      mergeMap((arr) => this.nativeWorker.completeActivityTask(arr.buffer.slice(arr.byteOffset))),
+      tap({ complete: () => this.log.debug('Activities complete') })
+    );
+  }
 
-    if (this.options.taskQueue === undefined) {
-      throw new Error('Worker taskQueue not defined');
-    }
-    const queueName = this.options.taskQueue;
+  protected takeUntilState<T>(state: State): MonoTypeOperatorFunction<T> {
+    return takeUntil(this.stateSubject.pipe(filter((value) => value === state)));
+  }
+
+  protected takeUntilIdle<T>(numInFlightActivitiesSubject: Observable<number>): MonoTypeOperatorFunction<T> {
```

Comment:
Oops, it was supposed to say activations not activities.
I'll fix in the next PR.